_posts/2019-11-09-using-vault-with-strimzi.md

```diff
@@ -0,0 +1,185 @@
+
+So how does Vault fit in with Strimzi and Kafka? Well, Vault has the concept of [`Secrets Engines`](https://www.vaultproject.io/docs/secrets/):
+
+> Secrets engines are components which store, generate, or encrypt data. Secrets engines are incredibly flexible, so it is easiest to think about them in terms of their function. Secrets engines are provided some set of data, they take some action on that data, and they return a result.\
+> Some secrets engines simply store and read data - like encrypted Redis/Memcached. Other secrets engines connect to other services and generate dynamic credentials on demand. Other secrets engines provide encryption as a service, totp generation, certificates, and much more.
+
+It is this last point, certificates, which we are going to be looking at. Specifically, we will be using the [Vault PKI Secrets Engine](https://www.vaultproject.io/docs/secrets/pki/index.html) to create a CA, which we will then install instead of using CA certificates and private keys generated by the [Strimzi Cluster Operator](http://strimzi.io/docs/master/#cluster-operator-str).
+
+# SSL, Strimzi & Security
+
+## The `incubator` Kafka charts
+
+Like me, you may have tried configuring SSL for Kafka using the `incubator` Helm charts (you can find them [here](https://github.com/helm/charts/tree/master/incubator/kafka)), and realised that in fact this is _far from trivial_. I had initially configured an `initContainer` to get certificates from the Vault PKI backend using the Vault CLI, which did indeed work. However, in addition to the startup scripts in the main broker container and another custom `initContainer` I had configured to set up rack awareness, this caused the the pod startup times to become quite long!
+
+At this point I had configured the brokers and clients with SSL, but what about the Zookeepers? Zookeeper does not support TLS, which is problematic if, for example, you are a financial organisation streaming sensitive data through your Kafka cluster. Even if you're not a fintech company, but you're a sensible individual with security concerns, this might be a deal breaker for you!
+
+## Strimzi & Security
+
+Straight out of the box, [Strimzi provides all of this for you](http://strimzi.io/docs/master/#security-str):
+
+> Strimzi supports encrypted communication between the Kafka and Strimzi components using the TLS protocol. Communication between Kafka brokers (interbroker communication), between Zookeeper nodes (internodal communication), and between these and the Strimzi operators is always encrypted. Communication between Kafka clients and Kafka brokers is encrypted according to how the cluster is configured. For the Kafka and Strimzi components, TLS certificates are also used for authentication.\
```

Comment:
the "\" here as well?

Reply:
It was just for formatting in the markdown preview. Will it cause issues when published?

Reply:
Maybe yes, let's delete this and if needed we'll fix after publication.

Reply:
Removing it.